src/lib/operators/join_sort_merge/radix_cluster_sort.hpp

```diff
@@ -345,10 +321,15 @@ class RadixClusterSort {
     ColumnMaterializer<T> right_column_materializer(!_equi_case, _materialize_null_right);
     auto materialization_left = left_column_materializer.materialize(_input_table_left, _left_column_id);
     auto materialization_right = right_column_materializer.materialize(_input_table_right, _right_column_id);
-    auto materialized_left_segments = std::move(materialization_left.first);
-    auto materialized_right_segments = std::move(materialization_right.first);
-    output.null_rows_left = std::move(materialization_left.second);
-    output.null_rows_right = std::move(materialization_right.second);
+    auto materialized_left_segments = std::move(std::get<0>(materialization_left));
+    auto materialized_right_segments = std::move(std::get<0>(materialization_right));
+    output.null_rows_left = std::move(std::get<1>(materialization_left));
+    output.null_rows_right = std::move(std::get<1>(materialization_right));
+    auto samples_left = std::get<2>(materialization_left);
+    auto samples_right = std::get<2>(materialization_right);
+
+    // Append right samples to left samples and sort
+    samples_left.insert(samples_left.end(), samples_right.begin(), samples_right.end());
```

Comment:
reserve?

Reply:
Not necessary here: https://stackoverflow.com/a/35359472/1147726

Shall I add a short comment?

Reply:
Learned something new. It'd add a comment, yes.

Reply:
Added a comment.